File: DSA/VedantChouhan_590024563/Day-22/q2.c
#include <stdio.h>

int main() {

    int cards[] = {3, 9, 1, 2};
    int n = sizeof(cards) / sizeof(cards[0]);

    int left = 0;
    int right = n - 1;

    int player1 = 0;
    int player2 = 0;
    int turn = 0;

    while (left <= right) {

        int picked;

        if (cards[left] > cards[right]) {
            picked = cards[left];
            left++;
        } else {
            picked = cards[right];
            right--;
        }

        if (turn == 0) {
            player1 += picked;
            turn = 1;
        } else {
            player2 += picked;
            turn = 0;
        }
    }

    printf("Player 1 Score: %d\n", player1);
    printf("Player 2 Score: %d\n", player2);

    return 0;
}

File: DSA/Navneet_590012555/Day22/Q2.java
class Solution {
    public int[] cardGame(int[] cards) {
        int l = 0, r = cards.length - 1;
        int p1 = 0, p2 = 0;
        boolean turn = true;

        while (l <= r) {
            int x;

            if (cards[l] >= cards[r])
                x = cards[l++];
            else
                x = cards[r--];

            if (turn)
                p1 += x;
            else
                p2 += x;

            turn = !turn;
        }

        return new int[] { p1, p2 };
    }
}